modules/basegl/src/shadercomponents/raycastingcomponent.cpp

```diff
@@ -127,21 +127,27 @@ result = drawISO(result, {isoparams}, {volume}Voxel[{channel}], {volume}VoxelPre
                  lighting, samplePosition, rayDirection,
                  cameraDir, rayPosition, rayStep, rayDepth);)");
 
-constexpr std::string_view classify1 = util::trim(R"(
-vec4 color{color} = texture({tf}, vec2({volume}Voxel[{channel}], 0.5));
+constexpr std::string_view init = util::trim(R"(
+ShadingParameters shadingParams = defaultShadingParameters();
+)");
+
+constexpr std::string_view colorInit = util::trim(R"(
+vec4 color{color} = vec4(0);
 )");
 
 constexpr std::string_view classify = util::trim(R"(
 color{color} = texture({tf}, vec2({volume}Voxel[{channel}], 0.5));
 )");
 
-constexpr std::string_view composite = util::trim(R"(
+constexpr std::string_view shadeAndComposite = util::trim(R"(
 if (color{color}.a > 0) {{
-    #if defined(SHADING_ENABLED) && defined(GRADIENTS_ENABLED)
-    vec3 worldSpacePosition = ({volume}Parameters.textureToWorld * vec4(samplePosition, 1.0)).xyz;
-    color{color}.rgb = APPLY_LIGHTING(lighting, color{color}.rgb, color{color}.rgb, vec3(1.0),
-                                      worldSpacePosition, -{gradient}, cameraDir);
+    shadingParams.colors = defaultMaterialColors(color{color}.rgb);
+    #if (defined(SHADING_ENABLED) && defined(GRADIENTS_ENABLED))
+    shadingParams.normal = -{gradient};
+    shadingParams.worldPosition = ({volume}Parameters.textureToWorld * vec4(samplePosition, 1.0)).xyz;
     #endif
+    color{color}.rgb = APPLY_LIGHTING_FUNC(lighting, shadingParams, cameraDir);
```

Comment:
No, since this would prevent this function from being called entirely. In case of the light volume calculations, these are encapsulated in `APPLY_LIGHTING_FUNC`.